### PR TITLE
validation: fix email valid

### LINKED ISSUE #2487: Email validation allows spaces in domain name
My environment:

```
├── Beego     : 1.7.2
├── GoVersion : go1.8
├── GOOS      : darwin
├── GOARCH    : amd64
├── NumCPU    : 4
├── GOPATH    : /Users/ak/Dev/go
├── GOROOT    : /usr/local/go
├── Compiler  : gc
└── Date      : Tuesday, 7 Mar 2017
```

I created the following code:

```
package main

import (
        "fmt"
        "github.com/astaxie/beego/validation"
)

func main() {
        email := "test@test.r u"
        valid := validation.Validation{}
        fmt.Printf("Validation of '%s': %v\n", email, valid.Email(email, "email").Ok)
}
```

and it gives:

```
$ go run main.go
Validation of 'test@test.r u': true
```

There seems to be a mistake in `emailPattern` at https://github.com/astaxie/beego/blob/master/validation/validators.go#L537

### PR DESCRIPTION
**Pull request by miraclesu**
fix #2487

### CODE REVIEW

File: validation/validation_test.go [hunk 1]
```diff
@@ -214,6 +214,12 @@ func TestEmail(t *testing.T) {
 	if !valid.Email("suchuangji@gmail.com", "email").Ok {
 		t.Error("\"suchuangji@gmail.com\" is a valid email address should be true")
 	}
+	if valid.Email("@suchuangji@gmail.com", "email").Ok {
+		t.Error("\"@suchuangji@gmail.com\" is a valid email address should be false")
+	}
+	if valid.Email("@suchuangji@gmail.com", "email").Ok {
```
Comment (astaxie) [inline validation/validation_test.go]: it looks this valid is the same as above.

Comment (miraclesu) [inline validation/validation_test.go]: just similar, this is start with `@`

Comment (astaxie) [inline validation/validation_test.go]: but it's the same as line 217

Comment (miraclesu) [inline validation/validation_test.go]: :sweat_smile: it's ok now

**Pull request merged**